test/support/encoded_schema.json

```diff
@@ -0,0 +1,472 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "contact": {},
+    "description": "Travel",
+    "license": {},
+    "title": "Duffel Technology Ltd.",
+    "version": "1.0.0"
+  },
+  "servers": [
+    {
+      "description": "Development server",
+      "url": "https://development.gigantic-server.com/v1",
+      "variables": {
+        "username": {
+          "default": "demo",
+          "description": "this value is assigned by the service provider, in this example `gigantic-server.com`"
+        }
+      }
+    }
+  ],
+  "extensions": {
```

Comment:
Is this exercising the [openapi extensions](https://swagger.io/docs/specification/openapi-extensions/) functionality?

If so, it should be something like:

```
“x-test”: true
```

Which would need to be detected and inserted into the `extensions` map - the reverse of what we do for encoding: https://github.com/open-api-spex/open_api_spex/blob/master/lib/open_api_spex/extendable.ex#L10

Reply:
Hey @mbuhot, I've made sure the test covers "x-test" now instead. Could you please clarify what you meant by:

> detected and inserted into the extensions map

According to the documentation many objects can have extensions e.g. https://swagger.io/specification/#contactObject

What is the desired behaviour, say for `OpenApiSpex.Contact`? This struct currently does not support extensions (as is the case for many other structs). Happy to open a ticket for this or address it in this pull-request here if you think it's the right place for it!

Reply:
@waltfy to clarify, extensions are pattern fields. Which means that they exist within the containing JSON object but don't have a pre-defined name in the schema. Extensions must be detected by the presence of the leading `x-` in the key name - the surrounding `extensions` map is added by `OpenApiSpex` to make it representable as an elixir struct.

Your example should be something like:

```json
{
   "openapi": "3.0.0",
   "info": {
     "description": "Travel",
     "title": "Duffel Technology Ltd.",
     "version": "1.0.0"
   },
   "servers": [
     {
       "description": "Development server",
       "url": "https://development.gigantic-server.com/v1",
       "variables": {
         "username": {
           "default": "demo",
           "description": "this value is assigned by the service provider, in this example `gigantic-server.com`"
         }
       }
     }
   ],
   "x-test": true
}
```

I think we could add another function to the `OpenApiSpex.Extendable` protocol to convert from a bare map to a struct:

```elixir
defprotocol OpenApiSpex.Extendable do
  @fallback_to_any true
  def to_map(struct) # Converts from struct with `extensions` field to map 
  def from_map(map) # Converts from map with "x-*" extension fields to struct with `extensions` field
end
```

Happy to move this to a separate issue, as I don't think we depend on any extensions for casting and validating requests.

Reply:
Yeah, I've realised this — I think I totally misunderstood how it worked before. Agreed, let's move into a separate issue.